src/http/server/response.cr

```diff
@@ -123,16 +123,23 @@ class HTTP::Server
       @output.closed?
     end
 
-    # Generates an error response using *message* and *code*.
+    # Sends an error response.
     #
-    # Calls `reset` and then writes the given message.
+    # Calls `#reset`, writes the given status, and closes the response.
+    def respond_with_error(status : HTTP::Status)
+      respond_with_error(status.description, status.code)
```

Comment:
Minor optimization: I'd use this method as the _main_ impl, since this way you could avoid allocating `HTTP::Status` twice.

Reply:
I thought about that, but `HTTP::Status` is an enum, so it's actually not an allocation, just a cast and some validation. In order to provide a custom error message, this would need an additional `message` argument as well. But this should be the second (optional) argument, which would be inverse to the existing method.
However, that might really be the better API.
Also, I figured this method should perhaps be renamed. After all, it is not only useful for errors, but can send any kind of status code.
So my suggestion would be to change the API to this:

```cr
def respond_with_status(status : HTTP::Status, message : String? = nil)
  message || = status.description
  # impl
end

def respond_with_status(status : Int, message : String? = nil)
  respond_with_status(HTTP::Status.new(status), message)
end
```

Reply:
Sounds like a nice improvement!